packages/shared/selectors/k8s.ts

```diff
@@ -0,0 +1,7 @@
+import { K8sResourceCommon } from '@openshift-console/dynamic-plugin-sdk';
+
+type GetStringProperty<T = K8sResourceCommon> = (resource: T) => string;
+
+export const getName: GetStringProperty = (resource) => resource.metadata?.name;
+
+export const getUID: GetStringProperty = (resource) => resource.metadata?.uid;
```

Comment:
if this is generic utility , then in usage resource can be undefined as well.

```suggestion
export const getName: GetStringProperty = (resource) => resource?.metadata?.name;

export const getUID: GetStringProperty = (resource) => resource?.metadata?.uid;
```